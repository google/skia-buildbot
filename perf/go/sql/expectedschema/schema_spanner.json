{
  "ColumnNameAndType": {
    "alerts.alert": "character varying def: nullable:YES",
    "alerts.config_state": "bigint def:'0'::bigint nullable:YES",
    "alerts.createdat": "timestamp with time zone def:CURRENT_TIMESTAMP nullable:YES",
    "alerts.id": "bigint def:nextval('Alerts_seq'::text) nullable:NO",
    "alerts.last_modified": "bigint def: nullable:YES",
    "alerts.sub_name": "character varying def: nullable:YES",
    "alerts.sub_revision": "character varying def: nullable:YES",
    "anomalygroups.action": "character varying def: nullable:YES",
    "anomalygroups.action_time": "timestamp with time zone def: nullable:YES",
    "anomalygroups.anomaly_ids": "ARRAY def: nullable:YES",
    "anomalygroups.bisection_id": "character varying def: nullable:YES",
    "anomalygroups.common_rev_end": "bigint def: nullable:YES",
    "anomalygroups.common_rev_start": "bigint def: nullable:YES",
    "anomalygroups.createdat": "timestamp with time zone def:CURRENT_TIMESTAMP nullable:YES",
    "anomalygroups.creation_time": "timestamp with time zone def:now() nullable:YES",
    "anomalygroups.culprit_ids": "ARRAY def: nullable:YES",
    "anomalygroups.group_meta_data": "jsonb def: nullable:YES",
    "anomalygroups.id": "character varying def:spanner.generate_uuid() nullable:NO",
    "anomalygroups.last_modified_time": "timestamp with time zone def: nullable:YES",
    "anomalygroups.reported_issue_id": "character varying def: nullable:YES",
    "commits.author": "character varying def: nullable:YES",
    "commits.commit_number": "bigint def: nullable:NO",
    "commits.commit_time": "bigint def: nullable:YES",
    "commits.createdat": "timestamp with time zone def:CURRENT_TIMESTAMP nullable:YES",
    "commits.git_hash": "character varying def: nullable:NO",
    "commits.subject": "character varying def: nullable:YES",
    "culprits.anomaly_group_ids": "ARRAY def: nullable:YES",
    "culprits.createdat": "timestamp with time zone def:CURRENT_TIMESTAMP nullable:YES",
    "culprits.group_issue_map": "jsonb def: nullable:YES",
    "culprits.host": "character varying def: nullable:YES",
    "culprits.id": "character varying def:spanner.generate_uuid() nullable:NO",
    "culprits.issue_ids": "ARRAY def: nullable:YES",
    "culprits.last_modified": "bigint def: nullable:YES",
    "culprits.project": "character varying def: nullable:YES",
    "culprits.ref": "character varying def: nullable:YES",
    "culprits.revision": "character varying def: nullable:YES",
    "favorites.createdat": "timestamp with time zone def:CURRENT_TIMESTAMP nullable:YES",
    "favorites.description": "character varying def: nullable:YES",
    "favorites.id": "character varying def:spanner.generate_uuid() nullable:NO",
    "favorites.last_modified": "bigint def: nullable:YES",
    "favorites.name": "character varying def: nullable:YES",
    "favorites.url": "character varying def: nullable:NO",
    "favorites.user_id": "character varying def: nullable:NO",
    "graphsshortcuts.createdat": "timestamp with time zone def:CURRENT_TIMESTAMP nullable:YES",
    "graphsshortcuts.graphs": "character varying def: nullable:YES",
    "graphsshortcuts.id": "character varying def: nullable:NO",
    "paramsets.createdat": "timestamp with time zone def:CURRENT_TIMESTAMP nullable:YES",
    "paramsets.param_key": "character varying def: nullable:NO",
    "paramsets.param_value": "character varying def: nullable:NO",
    "paramsets.tile_number": "bigint def: nullable:NO",
    "postings.createdat": "timestamp with time zone def:CURRENT_TIMESTAMP nullable:YES",
    "postings.key_value": "character varying def: nullable:NO",
    "postings.tile_number": "bigint def: nullable:NO",
    "postings.trace_id": "bytea def: nullable:NO",
    "regressions.alert_id": "bigint def: nullable:NO",
    "regressions.commit_number": "bigint def: nullable:NO",
    "regressions.createdat": "timestamp with time zone def:CURRENT_TIMESTAMP nullable:YES",
    "regressions.migrated": "boolean def: nullable:YES",
    "regressions.regression": "character varying def: nullable:YES",
    "regressions.regression_id": "character varying def: nullable:YES",
    "regressions2.alert_id": "bigint def: nullable:YES",
    "regressions2.cluster_summary": "jsonb def: nullable:YES",
    "regressions2.cluster_type": "character varying def: nullable:YES",
    "regressions2.commit_number": "bigint def: nullable:YES",
    "regressions2.createdat": "timestamp with time zone def:CURRENT_TIMESTAMP nullable:YES",
    "regressions2.creation_time": "timestamp with time zone def:now() nullable:YES",
    "regressions2.frame": "jsonb def: nullable:YES",
    "regressions2.id": "character varying def:spanner.generate_uuid() nullable:NO",
    "regressions2.is_improvement": "boolean def: nullable:YES",
    "regressions2.median_after": "real def: nullable:YES",
    "regressions2.median_before": "real def: nullable:YES",
    "regressions2.prev_commit_number": "bigint def: nullable:YES",
    "regressions2.triage_message": "character varying def: nullable:YES",
    "regressions2.triage_status": "character varying def: nullable:YES",
    "reversekeymap.createdat": "timestamp with time zone def:CURRENT_TIMESTAMP nullable:YES",
    "reversekeymap.modified_value": "character varying def: nullable:NO",
    "reversekeymap.original_value": "character varying def: nullable:YES",
    "reversekeymap.param_key": "character varying def: nullable:NO",
    "shortcuts.createdat": "timestamp with time zone def:CURRENT_TIMESTAMP nullable:YES",
    "shortcuts.id": "character varying def: nullable:NO",
    "shortcuts.trace_ids": "character varying def: nullable:YES",
    "sourcefiles.createdat": "timestamp with time zone def:CURRENT_TIMESTAMP nullable:YES",
    "sourcefiles.source_file": "character varying def: nullable:NO",
    "sourcefiles.source_file_id": "bigint def:nextval('SourceFiles_seq'::text) nullable:NO",
    "subscriptions.bug_cc_emails": "ARRAY def: nullable:YES",
    "subscriptions.bug_component": "character varying def: nullable:YES",
    "subscriptions.bug_labels": "ARRAY def: nullable:YES",
    "subscriptions.bug_priority": "bigint def: nullable:YES",
    "subscriptions.bug_severity": "bigint def: nullable:YES",
    "subscriptions.contact_email": "character varying def: nullable:YES",
    "subscriptions.createdat": "timestamp with time zone def:CURRENT_TIMESTAMP nullable:YES",
    "subscriptions.hotlists": "ARRAY def: nullable:YES",
    "subscriptions.is_active": "boolean def: nullable:YES",
    "subscriptions.name": "character varying def: nullable:NO",
    "subscriptions.revision": "character varying def: nullable:NO",
    "tracevalues.commit_number": "bigint def: nullable:NO",
    "tracevalues.createdat": "timestamp with time zone def:CURRENT_TIMESTAMP nullable:YES",
    "tracevalues.source_file_id": "bigint def: nullable:YES",
    "tracevalues.trace_id": "bytea def: nullable:NO",
    "tracevalues.val": "real def: nullable:YES",
    "userissues.commit_position": "bigint def: nullable:NO",
    "userissues.createdat": "timestamp with time zone def:CURRENT_TIMESTAMP nullable:YES",
    "userissues.issue_id": "bigint def: nullable:NO",
    "userissues.last_modified": "timestamp with time zone def:now() nullable:YES",
    "userissues.trace_key": "character varying def: nullable:NO",
    "userissues.user_id": "character varying def: nullable:NO"
  },
  "IndexNames": [
    "alerts.PRIMARY_KEY",
    "anomalygroups.PRIMARY_KEY",
    "commits.PRIMARY_KEY",
    "culprits.by_revision",
    "culprits.PRIMARY_KEY",
    "favorites.by_user_id",
    "favorites.PRIMARY_KEY",
    "graphsshortcuts.PRIMARY_KEY",
    "paramsets.by_tile_number",
    "paramsets.PRIMARY_KEY",
    "postings.by_trace_id",
    "postings.by_key_value",
    "postings.PRIMARY_KEY",
    "regressions.PRIMARY_KEY",
    "regressions2.by_commit_alert",
    "regressions2.by_alert_id",
    "regressions2.PRIMARY_KEY",
    "reversekeymap.PRIMARY_KEY",
    "shortcuts.PRIMARY_KEY",
    "sourcefiles.by_source_file",
    "sourcefiles.PRIMARY_KEY",
    "subscriptions.PRIMARY_KEY",
    "tracevalues.by_source_file_id",
    "tracevalues.PRIMARY_KEY",
    "userissues.PRIMARY_KEY"
  ]
}
