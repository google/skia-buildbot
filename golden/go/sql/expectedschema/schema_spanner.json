{
  "ColumnNameAndType": {
    "changelists.changelist_id": "character varying def: nullable:NO",
    "changelists.createdat": "timestamp with time zone def:CURRENT_TIMESTAMP nullable:YES",
    "changelists.last_ingested_data": "timestamp with time zone def: nullable:NO",
    "changelists.owner_email": "character varying def: nullable:NO",
    "changelists.status": "character varying def: nullable:NO",
    "changelists.subject": "character varying def: nullable:NO",
    "changelists.system": "character varying def: nullable:NO",
    "commitswithdata.commit_id": "character varying def: nullable:NO",
    "commitswithdata.createdat": "timestamp with time zone def:CURRENT_TIMESTAMP nullable:YES",
    "commitswithdata.tile_id": "bigint def: nullable:NO",
    "deprecatedexpectationundos.createdat": "timestamp with time zone def:CURRENT_TIMESTAMP nullable:YES",
    "deprecatedexpectationundos.expectation_id": "character varying def: nullable:NO",
    "deprecatedexpectationundos.id": "bigint def: nullable:NO",
    "deprecatedexpectationundos.ts": "timestamp with time zone def: nullable:NO",
    "deprecatedexpectationundos.user_id": "character varying def: nullable:NO",
    "deprecatedingestedfiles.createdat": "timestamp with time zone def:CURRENT_TIMESTAMP nullable:YES",
    "deprecatedingestedfiles.last_ingested": "timestamp with time zone def: nullable:NO",
    "deprecatedingestedfiles.source_file": "character varying def: nullable:NO",
    "deprecatedingestedfiles.source_file_id": "bytea def: nullable:NO",
    "diffmetrics.combined_metric": "real def: nullable:NO",
    "diffmetrics.createdat": "timestamp with time zone def:CURRENT_TIMESTAMP nullable:YES",
    "diffmetrics.dimensions_differ": "boolean def: nullable:NO",
    "diffmetrics.left_digest": "bytea def: nullable:NO",
    "diffmetrics.max_channel_diff": "bigint def: nullable:NO",
    "diffmetrics.max_rgba_diffs": "ARRAY def: nullable:NO",
    "diffmetrics.num_pixels_diff": "bigint def: nullable:NO",
    "diffmetrics.percent_pixels_diff": "real def: nullable:NO",
    "diffmetrics.right_digest": "bytea def: nullable:NO",
    "diffmetrics.ts": "timestamp with time zone def: nullable:NO",
    "expectationdeltas.createdat": "timestamp with time zone def:CURRENT_TIMESTAMP nullable:YES",
    "expectationdeltas.digest": "bytea def: nullable:NO",
    "expectationdeltas.expectation_record_id": "character varying def: nullable:NO",
    "expectationdeltas.grouping_id": "bytea def: nullable:NO",
    "expectationdeltas.label_after": "character varying def: nullable:NO",
    "expectationdeltas.label_before": "character varying def: nullable:NO",
    "expectationrecords.branch_name": "character varying def: nullable:YES",
    "expectationrecords.createdat": "timestamp with time zone def:CURRENT_TIMESTAMP nullable:YES",
    "expectationrecords.expectation_record_id": "character varying def:spanner.generate_uuid() nullable:NO",
    "expectationrecords.num_changes": "bigint def: nullable:NO",
    "expectationrecords.triage_time": "timestamp with time zone def: nullable:NO",
    "expectationrecords.user_name": "character varying def: nullable:NO",
    "expectations.createdat": "timestamp with time zone def:CURRENT_TIMESTAMP nullable:YES",
    "expectations.digest": "bytea def: nullable:NO",
    "expectations.expectation_record_id": "character varying def: nullable:YES",
    "expectations.grouping_id": "bytea def: nullable:NO",
    "expectations.label": "character varying def: nullable:NO",
    "gitcommits.author_email": "character varying def: nullable:NO",
    "gitcommits.commit_id": "character varying def: nullable:NO",
    "gitcommits.commit_time": "timestamp with time zone def: nullable:NO",
    "gitcommits.createdat": "timestamp with time zone def:CURRENT_TIMESTAMP nullable:YES",
    "gitcommits.git_hash": "character varying def: nullable:NO",
    "gitcommits.subject": "character varying def: nullable:NO",
    "groupings.createdat": "timestamp with time zone def:CURRENT_TIMESTAMP nullable:YES",
    "groupings.grouping_id": "bytea def: nullable:NO",
    "groupings.keys": "jsonb def: nullable:NO",
    "ignorerules.createdat": "timestamp with time zone def:CURRENT_TIMESTAMP nullable:YES",
    "ignorerules.creator_email": "character varying def: nullable:NO",
    "ignorerules.expires": "timestamp with time zone def: nullable:NO",
    "ignorerules.ignore_rule_id": "character varying def:spanner.generate_uuid() nullable:NO",
    "ignorerules.note": "character varying def: nullable:YES",
    "ignorerules.query": "jsonb def: nullable:NO",
    "ignorerules.updated_email": "character varying def: nullable:NO",
    "metadatacommits.commit_id": "character varying def: nullable:NO",
    "metadatacommits.commit_metadata": "character varying def: nullable:NO",
    "metadatacommits.createdat": "timestamp with time zone def:CURRENT_TIMESTAMP nullable:YES",
    "options.createdat": "timestamp with time zone def:CURRENT_TIMESTAMP nullable:YES",
    "options.keys": "jsonb def: nullable:NO",
    "options.options_id": "bytea def: nullable:NO",
    "patchsets.changelist_id": "character varying def: nullable:NO",
    "patchsets.commented_on_cl": "boolean def: nullable:NO",
    "patchsets.created_ts": "timestamp with time zone def: nullable:YES",
    "patchsets.createdat": "timestamp with time zone def:CURRENT_TIMESTAMP nullable:YES",
    "patchsets.git_hash": "character varying def: nullable:NO",
    "patchsets.patchset_id": "character varying def: nullable:NO",
    "patchsets.ps_order": "bigint def: nullable:NO",
    "patchsets.system": "character varying def: nullable:NO",
    "primarybranchdiffcalculationwork.calculation_lease_ends": "timestamp with time zone def: nullable:NO",
    "primarybranchdiffcalculationwork.createdat": "timestamp with time zone def:CURRENT_TIMESTAMP nullable:YES",
    "primarybranchdiffcalculationwork.grouping_id": "bytea def: nullable:NO",
    "primarybranchdiffcalculationwork.last_calculated_ts": "timestamp with time zone def: nullable:NO",
    "primarybranchparams.createdat": "timestamp with time zone def:CURRENT_TIMESTAMP nullable:YES",
    "primarybranchparams.key": "character varying def: nullable:NO",
    "primarybranchparams.tile_id": "bigint def: nullable:NO",
    "primarybranchparams.value": "character varying def: nullable:NO",
    "problemimages.createdat": "timestamp with time zone def:CURRENT_TIMESTAMP nullable:YES",
    "problemimages.digest": "character varying def: nullable:NO",
    "problemimages.error_ts": "timestamp with time zone def: nullable:NO",
    "problemimages.latest_error": "character varying def: nullable:NO",
    "problemimages.num_errors": "bigint def: nullable:NO",
    "secondarybranchdiffcalculationwork.branch_name": "character varying def: nullable:NO",
    "secondarybranchdiffcalculationwork.calculation_lease_ends": "timestamp with time zone def: nullable:NO",
    "secondarybranchdiffcalculationwork.createdat": "timestamp with time zone def:CURRENT_TIMESTAMP nullable:YES",
    "secondarybranchdiffcalculationwork.digests": "ARRAY def: nullable:NO",
    "secondarybranchdiffcalculationwork.grouping_id": "bytea def: nullable:NO",
    "secondarybranchdiffcalculationwork.last_calculated_ts": "timestamp with time zone def: nullable:NO",
    "secondarybranchdiffcalculationwork.last_updated_ts": "timestamp with time zone def: nullable:NO",
    "secondarybranchexpectations.branch_name": "character varying def: nullable:NO",
    "secondarybranchexpectations.createdat": "timestamp with time zone def:CURRENT_TIMESTAMP nullable:YES",
    "secondarybranchexpectations.digest": "bytea def: nullable:NO",
    "secondarybranchexpectations.expectation_record_id": "character varying def: nullable:NO",
    "secondarybranchexpectations.grouping_id": "bytea def: nullable:NO",
    "secondarybranchexpectations.label": "character varying def: nullable:NO",
    "secondarybranchparams.branch_name": "character varying def: nullable:NO",
    "secondarybranchparams.createdat": "timestamp with time zone def:CURRENT_TIMESTAMP nullable:YES",
    "secondarybranchparams.key": "character varying def: nullable:NO",
    "secondarybranchparams.value": "character varying def: nullable:NO",
    "secondarybranchparams.version_name": "character varying def: nullable:NO",
    "secondarybranchvalues.branch_name": "character varying def: nullable:NO",
    "secondarybranchvalues.createdat": "timestamp with time zone def:CURRENT_TIMESTAMP nullable:YES",
    "secondarybranchvalues.digest": "bytea def: nullable:NO",
    "secondarybranchvalues.grouping_id": "bytea def: nullable:NO",
    "secondarybranchvalues.options_id": "bytea def: nullable:NO",
    "secondarybranchvalues.secondary_branch_trace_id": "bytea def: nullable:NO",
    "secondarybranchvalues.source_file_id": "bytea def: nullable:NO",
    "secondarybranchvalues.tryjob_id": "character varying def: nullable:YES",
    "secondarybranchvalues.version_name": "character varying def: nullable:NO",
    "sourcefiles.createdat": "timestamp with time zone def:CURRENT_TIMESTAMP nullable:YES",
    "sourcefiles.last_ingested": "timestamp with time zone def: nullable:NO",
    "sourcefiles.source_file": "character varying def: nullable:NO",
    "sourcefiles.source_file_id": "bytea def: nullable:NO",
    "tiledtracedigests.createdat": "timestamp with time zone def:CURRENT_TIMESTAMP nullable:YES",
    "tiledtracedigests.digest": "bytea def: nullable:NO",
    "tiledtracedigests.grouping_id": "bytea def: nullable:NO",
    "tiledtracedigests.tile_id": "bigint def: nullable:NO",
    "tiledtracedigests.trace_id": "bytea def: nullable:NO",
    "traces.corpus": "character varying def: nullable:NO",
    "traces.createdat": "timestamp with time zone def:CURRENT_TIMESTAMP nullable:YES",
    "traces.grouping_id": "bytea def: nullable:NO",
    "traces.keys": "jsonb def: nullable:NO",
    "traces.matches_any_ignore_rule": "boolean def: nullable:YES",
    "traces.trace_id": "bytea def: nullable:NO",
    "tracevalues.commit_id": "character varying def: nullable:NO",
    "tracevalues.createdat": "timestamp with time zone def:CURRENT_TIMESTAMP nullable:YES",
    "tracevalues.digest": "bytea def: nullable:NO",
    "tracevalues.grouping_id": "bytea def: nullable:NO",
    "tracevalues.options_id": "bytea def: nullable:NO",
    "tracevalues.shard": "bigint def: nullable:NO",
    "tracevalues.source_file_id": "bytea def: nullable:NO",
    "tracevalues.trace_id": "bytea def: nullable:NO",
    "trackingcommits.createdat": "timestamp with time zone def:CURRENT_TIMESTAMP nullable:YES",
    "trackingcommits.last_git_hash": "character varying def: nullable:NO",
    "trackingcommits.repo": "character varying def: nullable:NO",
    "tryjobs.changelist_id": "character varying def: nullable:NO",
    "tryjobs.createdat": "timestamp with time zone def:CURRENT_TIMESTAMP nullable:YES",
    "tryjobs.display_name": "character varying def: nullable:NO",
    "tryjobs.last_ingested_data": "timestamp with time zone def: nullable:NO",
    "tryjobs.patchset_id": "character varying def: nullable:NO",
    "tryjobs.system": "character varying def: nullable:NO",
    "tryjobs.tryjob_id": "character varying def: nullable:NO",
    "valuesathead.corpus": "character varying def: nullable:NO",
    "valuesathead.createdat": "timestamp with time zone def:CURRENT_TIMESTAMP nullable:YES",
    "valuesathead.digest": "bytea def: nullable:NO",
    "valuesathead.grouping_id": "bytea def: nullable:NO",
    "valuesathead.keys": "jsonb def: nullable:NO",
    "valuesathead.matches_any_ignore_rule": "boolean def: nullable:YES",
    "valuesathead.most_recent_commit_id": "character varying def: nullable:NO",
    "valuesathead.options_id": "bytea def: nullable:NO",
    "valuesathead.trace_id": "bytea def: nullable:NO"
  },
  "IndexNames": [
    "changelists.system_status_ingested_idx",
    "changelists.status_ingested_idx",
    "changelists.PRIMARY_KEY",
    "commitswithdata.PRIMARY_KEY",
    "diffmetrics.PRIMARY_KEY",
    "expectationdeltas.PRIMARY_KEY",
    "expectationrecords.branch_ts_idx",
    "expectationrecords.PRIMARY_KEY",
    "expectations.label_idx",
    "expectations.PRIMARY_KEY",
    "gitcommits.commit_idx",
    "gitcommits.PRIMARY_KEY",
    "groupings.PRIMARY_KEY",
    "ignorerules.PRIMARY_KEY",
    "metadatacommits.PRIMARY_KEY",
    "options.PRIMARY_KEY",
    "patchsets.cl_order_idx",
    "patchsets.PRIMARY_KEY",
    "patchsets.IDX_patchsets_changelist_id_4B52682C0379EEF7",
    "primarybranchdiffcalculationwork.calculated_idx",
    "primarybranchdiffcalculationwork.PRIMARY_KEY",
    "primarybranchparams.PRIMARY_KEY",
    "problemimages.PRIMARY_KEY",
    "secondarybranchdiffcalculationwork.calculated_idx_1",
    "secondarybranchdiffcalculationwork.PRIMARY_KEY",
    "secondarybranchexpectations.PRIMARY_KEY",
    "secondarybranchparams.PRIMARY_KEY",
    "secondarybranchvalues.PRIMARY_KEY",
    "sourcefiles.PRIMARY_KEY",
    "tiledtracedigests.tile_trace_idx",
    "tiledtracedigests.grouping_digest_idx",
    "tiledtracedigests.PRIMARY_KEY",
    "tracevalues.trace_commit_idx",
    "tracevalues.PRIMARY_KEY",
    "traces.ignored_grouping_idx",
    "traces.grouping_ignored_idx",
    "traces.PRIMARY_KEY",
    "trackingcommits.PRIMARY_KEY",
    "tryjobs.cl_idx",
    "tryjobs.PRIMARY_KEY",
    "tryjobs.IDX_tryjobs_patchset_id_41BD54E7AE4FCE10",
    "tryjobs.IDX_tryjobs_changelist_id_3D37D47FECB38315",
    "valuesathead.ignored_grouping_idx_1",
    "valuesathead.corpus_commit_ignore_idx",
    "valuesathead.PRIMARY_KEY",
    "deprecatedingestedfiles.PRIMARY_KEY",
    "deprecatedexpectationundos.PRIMARY_KEY"
  ]
}
